http-client.ts

```diff
@@ -97,65 +95,77 @@ export class HttpClient implements Server.IHttpClient {
 
 			this.$logger.trace("httpRequest: %s", util.inspect(options));
 
-			let request = http.request(options, (response: Server.IRequestResponseData) => {
-				let data: string[] = [];
-				let isRedirect = helpers.isResponseRedirect(response);
-				let successful = helpers.isRequestSuccessful(response);
-				if (!successful) {
-					pipeTo = undefined;
-				}
-
-				let responseStream = response;
-				switch (response.headers["content-encoding"]) {
-					case "gzip":
-						responseStream = responseStream.pipe(zlib.createGunzip());
-						break;
-					case "deflate":
-						responseStream = responseStream.pipe(zlib.createInflate());
-						break;
-				}
-
-				if (pipeTo) {
-					pipeTo.on("finish", () => {
-						this.$logger.trace("httpRequest: Piping done. code = %d", response.statusCode.toString());
-						this.setResponseResult(promiseActions, timerId, { response });
-					});
-
-					pipeTo = this.trackDownloadProgress(pipeTo);
-
-					responseStream.pipe(pipeTo);
-				} else {
-					responseStream.on("data", (chunk: string) => {
-						data.push(chunk);
-					});
-
-					responseStream.on("end", () => {
-						this.$logger.trace("httpRequest: Done. code = %d", response.statusCode.toString());
-						let responseBody = data.join("");
-
-						if (successful || isRedirect) {
-							this.setResponseResult(promiseActions, timerId, { body: responseBody, response });
-						} else {
-							let errorMessage = this.getErrorMessage(response, responseBody);
-							let err: any = new Error(errorMessage);
-							err.response = response;
-							err.body = responseBody;
-							this.setResponseResult(promiseActions, timerId, { err });
-						}
-					});
-				}
-			});
-
-			request.on("error", (err: Error) => {
-				this.setResponseResult(promiseActions, timerId, { err });
-			});
+			options.url = options.url || `${options.proto}://${options.host}${options.path}`;
+			options.encoding = null;
+			options.followAllRedirects = true;
+
+			let requestObj = request(options);
```

Comment:
Is disputable because calling `.end()` alters the object's state but nonetheless point taken - will change it to `const`